CMakeLists.txt

```
@@ -537,8 +537,17 @@ endfunction()
 
 if(CMAKE_CUDA_COMPILER)
   if (NOT CUDA_TARGET_ARCHS)
-    # Volta, Ampere, Hopper
-    set(CUDA_TARGET_ARCHS  "70;80;90")
+    find_package(CUDAToolkit REQUIRED)
+
+    message(STATUS "Found CUDA Toolkit version: ${CUDAToolkit_VERSION}")
+
+    if (CUDAToolkit_VERSION VERSION_LESS 13.0)
+      # Ampere, Hopper
+      set(CUDA_TARGET_ARCHS  "80;90")
+    else()
+      # Ampere, Hopper, Blackwell
+      set(CUDA_TARGET_ARCHS  "80;90;100")
```

This would keep support for Turing.
```suggestion
      set(CUDA_TARGET_ARCHS  "75,80;90")
    else()
      # Ampere, Hopper, Blackwell
      set(CUDA_TARGET_ARCHS  "75,80;90;100")
```

---

add in bcdb26f64d